lib/src/ui/movies_page.dart

```diff
@@ -112,9 +85,46 @@ class _MoviesPageState extends State<MoviesPage> {
                     widget.bloc.fetchByMovieName(textField);
                   },
                 ),
-              ),
-            ],
+          leading: IconButton(
+            icon: Icon(
+              Icons.view_headline_outlined,
+              color: Colors.purple,
+              size: UiConstants.iconSize,
+            ),
+            onPressed: () {},
           ),
+          actions: [
+            isSearching
+                ? IconButton(
+                    icon: Icon(
+                      Icons.cancel,
+                      color: Colors.purple,
+                      size: UiConstants.iconSize,
+                    ),
+                    onPressed: () {
+                      setState(
+                        () {
+                          this.isSearching = false;
+                          widget.bloc.fetchTrendingMovies();
+                        },
+                      );
+                    },
+                  )
+                : IconButton(
+                    icon: Icon(
+                      Icons.search,
+                      color: Colors.purple,
+                      size: UiConstants.iconSize,
+                    ),
+                    onPressed: () {
+                      setState(
+                        () {
+                          this.isSearching = true;
+                        },
+                      );
+                    },
+                  ),
```

Comment:
You can wrap this inside a new widget, as parameter you can use isSearching and bloc

Reply:
This is not possible because isSearching's state is previously modified in the 'title' property and this variable needs that reference. I could wrap the whole functionality into a new Widget but it would be the same as the AppBar() itself. So I recommend to keep this logic in this file